platform/titanium/runtime/components/list-view/list-section.js

```diff
@@ -1,24 +1,65 @@
 export default {
 	name: 'ListSection',
-
 	props: {
 		items: {
-			type: Array
+			type: Array,
+			default: () => []
 		}
 	},
-
-	render(h) {
-		return h('titanium-list-section');
-	},
-
 	watch: {
-		items(newValue) {
-			this.$el.setAttribute('items', newValue);
+		items(newValue, oldValue) {
+			if (newValue !== oldValue) {
+				this._newDataSource = true;
+				this.$titaniumView.setItems(newValue);
+			}
 		}
 	},
-
+	render(h) {
+		const items = [];
+		if (this.items.length) {
+			for (let index = 0; index < this.items.length; index++) {
+				const item = this.items[index];
+				items.push(h('list-item', {
+					props: { index, item },
+					on: {
+						'item-updated': this.onItemUpdated
+					}
+				}));
+			}
+		}
+		return h('titanium-list-section', items);
+	},
+	created() {
+		this._newDataSource = false;
+	},
 	mounted() {
 		this.$el.setAttribute('items', this.items);
-		this.$parent.appendSection(this.$el.titaniumView);
+		this.$parent.appendSection(this.$titaniumView);
+	},
+	updated() {
+		this._newDataSource = false;
+	},
+	methods: {
+		insertItem(itemVm) {
+			if (this._newDataSource) {
+				return;
+			}
+
+			this.$titaniumView.insertItemsAt(itemVm.index, [itemVm.item]);
```

Comment:
Do we need to wrap methods like this or is it for convenience?

Reply:
They are just for reactive updates, that is, that when the `items` of the section are changed, the list view updates itself.

If a user wants to directly interact with the underlying Titanium view there is the special `$titaniumView` instance property that returns the root Titanium view of a Vue component.